io/driver/driver_parameter.py

```diff
@@ -0,0 +1,170 @@
+        self.sysfs_chk = self.params.get('sysfs_check_required', default=None)
+        if self.ifaces[0:2] == 'ib':
+            configure_network.set_ip(self.ipaddr, self.netmask, self.ifaces,
+                                     interface_type='Infiniband')
+        else:
+            configure_network.set_ip(self.ipaddr, self.netmask, self.ifaces,
+                                     interface_type='Ethernet')
+        self.load_unload_sleep_time = 30
+        self.error_modules = []
+        self.mod_list = []
+        self.uname = linux_modules.platform.uname()[2]
+
+    def built_in_module(self, module):
+        """
+        checking whether the given module is built_in module or not
+        """
+        path = "/lib/modules/%s/modules.builtin" % self.uname
+        for each in genio.read_all_lines(path):
+            out = process.getoutput(each.split('/')[-1])
+            if module == out.split('.'[0]):
+                return True
+            return False
+
+    def ping_check(self, options):
+        '''
+        Checks if the ping to peer works. Returns True if it works.
+        Returns False otherwise.
+        '''
+        cmd = "ping -I %s %s %s" % (self.ifaces, options, self.peer)
+        if process.system(cmd, shell=True, verbose=True,
+                          ignore_status=True) != 0:
+            return False
+        return True
+
+    def sysfs_value_check(self):
+        '''
+        Checks if sysfs value matches to test value, Returns True if yes.
+        Returns False otherwise.
+        '''
+        value_check = process.system_output('/usr/bin/cat \
+                               /sys/module/%s/parameters/%s' %
+                                            (self.module, self.param_name
+                                             )).decode('utf-8')
+        if self.param_value not in value_check:
+            return False
+        return True
+
+    def param_check(self):
+        '''
+        Checks if Param is available for the Module, Returns True if yes.
+        Returns False otherwise.
+        '''
+        value_check = process.system_output('/usr/sbin/modinfo -p %s '
+                                            % self.module).decode('utf-8')
+        if self.param_name not in value_check:
+            return False
+        return True
+
+    def module_load_unload(self, mod1):
+        """
+        Unloading and loading the given module
+        """
+        if linux_modules.module_is_loaded(mod1) is False:
+            linux_modules.load_module(mod1)
+            time.sleep(self.load_unload_sleep_time)
+        sub_mod = linux_modules.get_submodules(mod1)
+        if sub_mod:
+            for mod in sub_mod.split(' '):
+                linux_modules.unload_module(mod)
+                if linux_modules.module_is_loaded(mod) is True:
+                    self.error_modules.append(mod)
+                    break
+        cmd = "rmmod %s" % mod1
```

Comment:
Please use https://github.com/avocado-framework/avocado/blob/master/avocado/utils/linux_modules.py#L162

Reply:
There is an catch.

unload_module uses the modinfo -r method of removing....    If this is being used.. We cannot use insmod for sending in parameters.  

The actual way of working is
#rmmod driver_name
#insmod driver_name

or

#modprobe -r driver_name
#modprobe driver_name

btw even modinfo driver_name <parameter>=<value> wont work...   It has to be rmmod and then insmod.

Reply:
What is the reason `modprobe driver_name parameter=value` won't work @manvanthar ? Because, that is the ideal way

Reply:
@narasimhan-v ... Here is the output... Modprobe is mainly used to pick the default parameters.

# modprobe -r mlx5_ib
# modprobe mlx5_ib dc_cnak_qp_depth=1500
# cat /sys/module/mlx5_ib/parameters/dc_cnak_qp_depth 
8192

Reply:
```
# cat /sys/module/kvm_hv/parameters/one_vm_per_core 
N
# rmmod kvm_hv
# modprobe kvm_hv one_vm_per_core=Y
# cat /sys/module/kvm_hv/parameters/one_vm_per_core 
Y
#
```

Reply:
hmm current libraries uses modprobe -r.    Should that be changed to rmmod?

Reply:
`modprobe -r` or `rmmod` is fine. I was just demonstrating `modprobe module param=value` works fine.

Reply:
Changes included in latest code!!